### PR TITLE
test(outputs.sql): do not write to file during testing

### PR DESCRIPTION
**Pull request by powersj**
Instead, use the output of the commands and compare that to the expected values. Due to some extra encoding bytes, some need to be stripped before comparing so each metric was split out into different files.

### CODE REVIEW

File: plugins/outputs/sql/sql_test.go [hunk 1]
```diff
@@ -211,31 +211,32 @@ func TestMysqlIntegration(t *testing.T) {
...
-		return !os.IsNotExist(err)
-	}, 5*time.Second, 10*time.Millisecond)
+	cases := []struct {
+		expected string
+	}{
+		{"INSERT INTO `metric_one` VALUES ('2021-05-17 22:04:45','tag1','tag2',1234,2345,1,0,1000000000,3.1415);"},
+		{"INSERT INTO `metric_two` VALUES ('2021-05-17 22:04:45','tag3','string1');"},
+		{"INSERT INTO `metric three` VALUES ('2021-05-17 22:04:45','tag4','string2');"},
+	}
+	for _, tc := range cases {
```
Comment (reimda) [inline plugins/outputs/sql/sql_test.go:221]: This table test doesn't look related to switching from file to reading stdout from the exec. Did it get migrated from somewhere else like a data file? Do we need to remove it from somewhere else?

**Review by reimda — COMMENTED** [review]
Switching from writing a file to reading stdout from the exec seems like a safe move

File: plugins/outputs/sql/sql_test.go [hunk 2]
```diff
@@ -211,31 +211,35 @@ func TestMysqlIntegration(t *testing.T) {
...
+			rc, out, err := container.Exec([]string{
+				"bash",
+				"-c",
+				"mariadb-dump --user=" + username +
+					" --password=" + password +
+					" --compact --skip-opt " +
+					dbname,
+			})
+			require.NoError(t, err)
+			require.Equal(t, 0, rc)
```
Comment (srebhan) [inline plugins/outputs/sql/sql_test.go:232]: Do you really need this in the `Eventually` block as it will get executed every 10 milliseconds... Same for reading of the expected file...

File: plugins/outputs/sql/sql_test.go [hunk 3]
```diff
@@ -295,38 +299,35 @@ func TestPostgresIntegration(t *testing.T) {
...
+				// blank lines.
+				"|grep -E -v '(^--|^$)'",
+		})
+		require.NoError(t, err)
+		require.Equal(t, 0, rc)
+
+		bytes, err := io.ReadAll(out)
+		require.NoError(t, err)
+
+		return strings.Contains(string(bytes), string(expected))
```
Comment (srebhan) [inline plugins/outputs/sql/sql_test.go:329]: Same question as above.

**Review by srebhan — COMMENTED** [review]
Thanks for the effort @powersj! I have two spots where putting command execution (and file reading) into the `require.Eventually` block which makes them getting executed in 10ms intervals...

File: plugins/outputs/sql/sql_test.go (hunk 2 above)
Comment (powersj) [reply plugins/outputs/sql/sql_test.go:232]: I too was hesitant to leave this in, but given the issues we have had with these integration tests I wanted to give this test every opportunity to wait for the database output. Some of the test failures made it seem that some of the data was just not present from the output, hence I left it in.

Comment (powersj) [reply plugins/outputs/sql/sql_test.go:232]: As far as the 10 milliseconds, that is what I have been using for most of these require eventually blocks: 5 second timeout and 10 millisecond retry. I'm happy to change this, but my goal was let this finish fast on fast systems and still retry on the slow circle ci system

Comment (srebhan) [reply plugins/outputs/sql/sql_test.go:232]: Can we then please move at least the reading of the expectation file out and increase the polling time to e.g. one second?
